src/controllers/observation-controller.ts

```diff
@@ -60,26 +51,46 @@ export default class ObservationController extends AbstractController {
   };
 
   private addObservation = async (req: Request, res: Response, next: NextFunction) => {
-    const newObservation = req.body;
-    if (!req.user) {
-      return res.status(401).send();
+    const rawObservation = req.body;
+    if (!req.user) res.status(401).send();
+
+    try {
+      const newObservation = await Observation.create({ ...rawObservation, finder: req.user.id });
+      await this.validate(newObservation);
+      const result = await this.observations.save(newObservation);
+      res.json(result);
+    } catch (e) {
+      next(e);
     }
+  };
 
-    // validation of Observation fields should be somewhere here
+  private findObservation = async (req: RequestWithObservation, res: Response, next: NextFunction) => {
+    const { observation }: { observation: Observation } = req;
+    try {
+      res.json(observation);
+    } catch (e) {
+      next(e);
+    }
+  };
 
+  private editObservation = async (req: RequestWithObservation, res: Response, next: NextFunction) => {
+    const { observation }: { observation: Observation } = req;
+    const rawObservation = req.body;
     try {
-      const observation = await Observation.create({ ...newObservation, finder: req.user.id });
-      const result = await this.observations.save(observation);
-      return res.json(result);
+      const updatedObservation = await this.observations.merge(observation, rawObservation);
+      await this.validate(rawObservation);
```

Comment:
As I remember this method can get 2 params when you update smth.

Reply:
Fixed.